src/app/screens/Send.tsx

```diff
@@ -63,7 +75,7 @@ function Send() {
             qrbox={200}
             qrCodeSuccessCallback={(decodedText) => {
               if (invoice !== decodedText) {
-                setInvoice(decodedText);
+                checkInvoice(decodedText);
```

Comment:
I would name this function `checkBip21AndSetInvoice`, so it's more clear what this actually does.

Reply:
I would even do something like: 

`setInvoice(extractInvoiceFrom(decodedText))`

this way the `extractInvoiceFrom` does not need to know anything about context where it is used and just returns the invoice string if it found one. 
Potentially this makes it easier to extract and reuse somewhere else and also easier to test.

Reply:
Thanks @bumi! I'll make these changes

Reply:
(not sure if it's a good idea, was just a thought. :D )
thnks